app/team.go

```diff
@@ -1222,14 +1222,36 @@ func (a *App) prepareInviteNewUsersToTeam(teamID, senderId string) (*model.User,
 		close(uchan)
 	}()
 
+	var channels []*model.Channel
+	if memberInvite != nil {
+		cchan := make(chan store.StoreResult, 1)
+		go func() {
+			channels, err := a.Srv().Store.Channel().GetChannelsByIds(memberInvite.Channels, false)
+			cchan <- store.StoreResult{Data: channels, NErr: err}
+			close(cchan)
+		}()
+
+		result := <-cchan
+		if result.NErr != nil {
+			return nil, nil, nil, model.NewAppError("prepareInviteNewUsersToTeam", "app.channel.get_channels_by_ids.app_error", nil, result.NErr.Error(), http.StatusInternalServerError)
+		}
+		channels := result.Data.([]*model.Channel)
```

Comment:
having **one** go routine immediately followed by a blocking channel read seems unnecessary.

I'm pretty sure this will produce the same result:, and might even be faster
```suggestion
		channels, err := a.Srv().Store.Channel().GetChannelsByIds(memberInvite.Channels, false)
		err != nil {
			return nil, nil, nil, model.NewAppError("prepareInviteNewUsersToTeam", "app.channel.get_channels_by_ids.app_error", nil, result.NErr.Error(), http.StatusInternalServerError)
		}
```

Reply:
going to give it a try.